src/library_wasi.js

```diff
@@ -76,7 +76,7 @@ var WasiLibrary = {
 
   args_sizes_get__sig: 'iii',
   args_sizes_get: function(pargc, pargv_buf_size) {
-#if MAIN_READS_PARAMS
+#if MAIN_READS_PARAMS && STANDALONE_WASM
```

Comment:
this change confuses me. do we not use `args_sizes_get` in non-standalone builds?

Reply:
In normal builds argv flows in from `callMain` through that arguments to main, and there is no `mainArgs` global at all (which is why this ifdef is needed.

In STANDALONE_WASM builds we call `_start` in `callMain` and stash argv in the `mainArgs` global so that it can be later read by `args_sizes_get`.    

Maybe one day we can unify these.. but that is how it works today.

Reply:
I see, sgtm.

Reply:
Actually, wait, then this function is never called in non-standalone builds, right? Then can't the standalone ifdef go on the entire function?

Reply:
We could yes.   Do you think that is better?   Since library functions are only included in demand I'm not sure it makes much difference.

Reply:
Its a slightly bigger/riskier change since it makes programs that depend on these symbols fail to link .. when today they would link fine and only fail if the functions were called.

Reply:
Failing to link seems better though? As it is definitely broken if `args_*` is called in nonstandalone mode, I think we should error eagerly on that.

That will help `FULL_LIBRARY` as well - these functions are not relevant for standalone mode, and should not be included there at all.

Reply:
OK fair enough.